src/canvas/Renderer.tsx

```diff
@@ -127,17 +146,37 @@ export class Renderer implements IRenderer {
 
         this._canvas.onmousedown = null;
         this._canvas.onmousemove = null;
-    }
+    };
 
-    private _endTracing = (position: Point): void => {
-        if (this._currPath) {
-            this._currPath.addPoint(position); 
-            this._currPath.snapEndpoints(this._document.getPatternPaths());
-            this._currPath.setFittedSegment();
+    /* Checks if the path starts by intersecting another path. If it does, and that intersection is 
+     * near the start of the intersected path, then the path start is snapped to the start of the 
+     * intersected path. If that intersection is near the end of the intersecte path, then the path start
+     * is snapped to the end of the intersected path. If the path starts near a point along the 
+     * intersected path, then the path start is snapped to that point along the intersected path. 
+     * If the intersected path is crossed at a point along the path and not an endpoint, the
+     * intersected path is bisected at the intersection point and the original path is replaced with
+     * the two new paths in the document. */
+    private _checkPathStartIntersectionAndSplit = (path: PatternPath, paths: PatternPath[]): void => {
+        const intersection = PathIntersection.findPathStartIntersectAlongPatternPath(path, paths);
+        if (intersection) {
+            path.snapStartToPoint(intersection.point);
```

Comment:
It looks like this line is repeated at line 177

Reply:
Thanks! I removed the first instance of it.